pkg/alertmanager/alertmanager_metrics.go

```diff
@@ -296,10 +296,10 @@ func (m *alertmanagerMetrics) Collect(out chan<- prometheus.Metric) {
 	data.SendSumOfCountersPerUser(out, m.alertsReceived, "alertmanager_alerts_received_total")
 	data.SendSumOfCountersPerUser(out, m.alertsInvalid, "alertmanager_alerts_invalid_total")
 
-	data.SendSumOfCountersPerUserWithLabels(out, m.numNotifications, "alertmanager_notifications_total", "integration")
-	data.SendSumOfCountersPerUserWithLabels(out, m.numFailedNotifications, "alertmanager_notifications_failed_total", "integration")
-	data.SendSumOfCountersPerUserWithLabels(out, m.numNotificationRequestsTotal, "alertmanager_notification_requests_total", "integration")
-	data.SendSumOfCountersPerUserWithLabels(out, m.numNotificationRequestsFailedTotal, "alertmanager_notification_requests_failed_total", "integration")
+	data.SendSumOfCountersPerUserWithLabelsAndOptions(out, m.numNotifications, "alertmanager_notifications_total", []string{"integration"}, util.SkipZeroValueMetrics)
```

Comment:
On first sight "AndOptions" looked like premature generalization to me, and I was going to suggest to generalize only when there are at least three variants in the future.

But there are already 6! :)

```
SendSumOfCounters
SendSumOfCountersPerUser
SendSumOfCountersPerUserWithLabels
SendSumOfCountersPerUserWithLabelsAndOptions
SendSumOfCountersWithLabels
```

Perhaps we should only have `SendSumOfCounters` and variants should really be "options"?

(But that's for another PR)

Reply:
_Looking into it._